src/main/java/serene/Serene.java

```diff
@@ -0,0 +1,402 @@
+        String marker = savedTask.substring(SAVED_INDEX_IS_DONE, SAVED_INDEX_IS_DONE + 1);
+        // Extract description
+        String descriptionAndTime = savedTask.substring(SAVED_INDEX_DESCRIPTION);
+        int timeIndex;
+        String description;
+        switch(taskType) {
+        case "T":
+            ToDo todo = new ToDo(descriptionAndTime);
+            if (marker.equals("X")) {
+                todo.markDone();
+            }
+            taskList.add(todo);
+            break;
+        case "D":
+            timeIndex = descriptionAndTime.indexOf(" (by: ");
+            description = descriptionAndTime.substring(0, timeIndex);
+            String by = descriptionAndTime.substring(timeIndex + TIME_OFFSET, descriptionAndTime.length() - 1);
+            Deadline deadline = new Deadline(description, by);
+            if (marker.equals("X")) {
+                deadline.markDone();
+            }
+            taskList.add(deadline);
+            break;
+        case "E":
+            timeIndex = descriptionAndTime.indexOf(" (at: ");
+            description = descriptionAndTime.substring(0, timeIndex);
+            String at = descriptionAndTime.substring(timeIndex + TIME_OFFSET, descriptionAndTime.length() - 1);
+            Event event = new Event(description, at);
+            if (marker.equals("X")) {
+                event.markDone();
+            }
+            taskList.add(event);
+            break;
+        }
+    }
+
+    private static void printWelcomeMessage() {
+        String logo = " #####  ####### ######  ####### #     # ####### \n"
+                + "#     # #       #     # #       ##    # #       \n"
+                + "#       #       #     # #       # #   # #       \n"
+                + " #####  #####   ######  #####   #  #  # #####   \n"
+                + "      # #       #   #   #       #   # # #       \n"
+                + "#     # #       #    #  #       #    ## #       \n"
+                + " #####  ####### #     # ####### #     # ####### ";
+        String greetLine = "Hello~ I'm Serene" + System.lineSeparator() + "What can I do for you?";
+        System.out.println(PARTITION_LINE);
+        System.out.println("Booting up");
+        System.out.println(logo);
+        printWithPartition(greetLine);
+    }
+
+    private static void printWithPartition(String input) {
+        System.out.println(PARTITION_LINE);
+        System.out.println(input);
+        System.out.println(PARTITION_LINE);
+    }
+
+    private static void operateSerene(Scanner in) {
+        while (statusOfSerene != DONE) {
+            String userInput = in.nextLine();
+            statusOfSerene = parseInput(userInput);
+        }
+    }
+
+    private static int parseInput(String userInput) {
+        // Split keyword from the rest of the input
+        String[] responsePartition = userInput.split(" ", 2);
+        String keyword = responsePartition[RESPONSE_INDEX_KEYWORD];
+        int operationState = CONTINUE;
```

Comment:
How about ensuring that the [level of abstraction](https://nus-cs2113-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard) here is same as below?